src/components/utils/useBoolean.ts

```diff
@@ -0,0 +1,17 @@
+import {useCallback, useState} from 'react';
+
+type SetTrue = () => void;
+type SetFalse = () => void;
+type Toggle = () => void;
+
+export function useBoolean(
+    initialState: boolean | (() => boolean),
+): [boolean, SetTrue, SetFalse, Toggle] {
+    const [value, setValue] = useState<boolean>(initialState);
+    return [
+        value,
+        useCallback(() => setValue(true), []),
+        useCallback(() => setValue(false), []),
+        useCallback(() => setValue((val) => !val), []),
```

Comment:
Aren't we need `val` as a dependency here?

Reply:
No, we aren't. `toggle` callback should not be recreated when the state changes